addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -778,9 +29,16 @@ publicWidget.registry.websiteSlidesUpload = publicWidget.Widget.extend({
     //--------------------------------------------------------------------------
 
     _openDialog: function ($element) {
-        return new SlideUploadDialog(this, $element.data()).open();
+        const dataset = $element.data();
+        this.call("dialog", "add", SlideUploadDialog, {
+            categoryId: dataset.categoryId,
+            channelId: dataset.channelId,
+            canPublish: dataset.canPublish === "True",
+            canUpload: dataset.canUpload === "True",
+            modulesToInstall: dataset.modulesToInstall ? JSON.parse(dataset.modulesToInstall?.replace(/'/g, '"')) : [],
```

Comment:
this is not correct, for example if we have `{'A': 'a\'b'}`
maybe the issue is that we "toString" the array instead of doing JSON.stringify ?

I think in `odoo/addons/website_slides/controllers/main.py` we can give the JSON.dumps instead of the array